Apps/JobScheduler/src/Startup.cs

```diff
@@ -130,7 +149,58 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 },
             });
 
-            app.UseMvc();
+            // This is needed if running behind a reverse proxy
+            // like ngrok which is great for testing while developing
+            app.UseForwardedHeaders(new ForwardedHeadersOptions
+            {
+                RequireHeaderSymmetry = false,
+                ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto,
+            });
+
+            app.UseCors();
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute("default", "{controller=JobScheduler}/{action=Index}/{id?}");
+            });
+        }
+
+        /// <summary>
+        /// This sets up the OIDC authentication for Hangfire.
+        /// </summary>
+        /// <param name="services">The passed in IServiceCollection.</param>
+        private void ConfigureAuthentication(IServiceCollection services)
+        {
+            string authorityEndPoint = this.configuration.GetValue<string>("OpenIdConnect:Authority");
+            bool requireHttpsMetadata = this.configuration.GetValue<bool>("OpenIdConnect:RequireHttpsMetadata");
+            string clientId = this.configuration.GetValue<string>("OpenIdConnect:ClientId");
+            string clientSecret = this.configuration.GetValue<string>("OpenIdConnect:ClientSecret");
+
+            services.AddAuthentication(auth =>
+            {
+                auth.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                auth.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+            })
+            .AddCookie(options =>
+            {
+                options.Cookie.Name = "HealthGateway_JobScheduler";
+            })
+            .AddOpenIdConnect(options =>
+            {
+                options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
```

Comment:
You can simplify this logic by binding to configuration and making all of the configurable with defaults.              
  this.configuration.GetSection("OpenIdConnect").Bind(options);

Reply:
+1

Reply:
yes, i know, but i hat debugging the magical binder. i like to literally see what is being set. so it looks deliberate.  Bind() requires the user to go on a goose chase to see what defaults are set, etc... i am happy to use the Bind() but just not a fan of it.

Reply:
I'm fine although we can chat when I'm back :)  I find one break point and I can confirm all of the bindings.